### PR TITLE
fix: close oidc providers per lib recommendation

### PR DESCRIPTION
**Pull request by markphelps**
While looking into #2571 , I noticed that we weren't calling `Close` on the OIDC provider as recommended by the `cap` library: https://pkg.go.dev/github.com/hashicorp/cap@v0.4.0/oidc#NewProvider

This change calls `Close` when we are done with the provider to release 'resources'

### CODE REVIEW

File: internal/server/auth/method/oidc/server.go [hunk 1]
```diff
@@ -192,6 +194,7 @@ func (s *Server) providerFor(provider string, state string) (*capoidc.Provider,
 
 	p, err := capoidc.NewProvider(config)
 	if err != nil {
+		p.Done()
```
Comment (erka) [inline internal/server/auth/method/oidc/server.go:197]: this probably will panic because `p` is nil.

Comment (markphelps) [reply internal/server/auth/method/oidc/server.go:197]: Ah good call. Dunno what I was thinking

File: internal/server/auth/method/oidc/server.go
Comment (erka) [inline internal/server/auth/method/oidc/server.go:1]: `Server.AuthorizeURL` also uses `providerFor`. It should have `defer provider.Done()` too.

Comment (markphelps) [reply internal/server/auth/method/oidc/server.go:1]: 🙌🏻 legend

**Review by GeorgeMac — APPROVED** [review]

**Pull request merged**